src/test/java/com/list/todo/services/TodoListServiceTest.java

```diff
@@ -69,69 +68,150 @@ public void getTodoListById_GetExistentTodoList_OptionalOfTodoList() {
         Assert.assertEquals(todoListPage, returnedTodoLists);
+        verify(tagTaskKeyService, times(1)).getTasksByTags(tagIds, USER_ID);
+        verify(todoListRepository, times(1)).findByCreatedByAndTodoListStatus(USER_USERNAME, TodoListStatus.ACTIVE, pageable);
     }
 
+
     @Test
     public void addTodoList_Successful_OptionalOfAddedTodoList() {
         //arrange
-        TodoList todoList = new TodoList();
-        todoList.setTodoListName(TODO_LIST_NAME);
-        todoList.setTodoListStatus(TodoListStatus.ACTIVE);
-        
-        when(todoListRepository.save(todoList)).thenReturn(todoList);
+        User user = new User();
+        TodoList todoList = createTodoList();
+        Set<Task> tasks = createSetOfTasks();
+        todoList.setTasks(tasks);
+
+        when(todoListRepository.save(any(TodoList.class))).thenReturn(todoList);
+        when(userService.getUserById(USER_ID)).thenReturn(Optional.of(user));
 
         //act
         Optional<TodoList> addedTodoList = todoListService.addTodoList(
-                new TodoListInput(todoList.getTodoListName(), TODO_LIST_COMMENT, new LinkedHashSet<>()), USER_ID);
+                new TodoListInput(todoList.getTodoListName(), TODO_LIST_COMMENT, tasks), USER_ID);
 
         //assert
-        verify(userService).getUserById(USER_ID);
+        verify(userService, times(1)).getUserById(USER_ID);
+        verify(todoListRepository, times(2)).save(any(TodoList.class));
+        verify(notificationService, times(1)).notifyAboutAddingTodoList(eq(user), any(TodoList.class));
         Assert.assertEquals(addedTodoList, Optional.of(todoList));
     }
 
     @Test
     public void updateTodoList_Successful_OptionalOfUpdatedTodoList() {
         //arrange
-        String newTodoListName = TODO_LIST_NAME_2;
-        TodoList todoList = Mockito.mock(TodoList.class);
+        User user = new User();
+        Set<Task> tasks = createSetOfTasks();
+        TodoList todoList = spy(createTodoList());
 
         when(todoListRepository.findById(TODO_LIST_ID)).thenReturn(Optional.of(todoList));
+        when(todoListRepository.save(todoList)).thenReturn(todoList);
+        when(userService.getUserById(USER_ID)).thenReturn(Optional.of(user));
 
         //act
         todoListService.updateTodoList(
-                TODO_LIST_ID, new TodoListInput(newTodoListName, TODO_LIST_COMMENT, new LinkedHashSet<>()), USER_ID);
+                TODO_LIST_ID, new TodoListInput(TODO_LIST_NAME, TODO_LIST_COMMENT, tasks), USER_ID);
 
         //assert
-        verify(todoList).setTodoListName(newTodoListName);
+        verify(todoList, times(1)).setTodoListName(TODO_LIST_NAME);
+        verify(todoList, times(1)).setComment(TODO_LIST_COMMENT);
+        verify(todoListRepository, times(1)).save(todoList);
+        verify(notificationService, times(1)).notifyAboutUpdatingTodoList(eq(user), any(TodoList.class));
     }
 
     @Test
     public void getTodoListsByUser_getAllTodoListsByExistentUser_ListOfTodoListsByUser() {
```

Comment:
`_ListOfTodoListsByUser` `By` preposition does not suit here. Don't afraid to be a bit more verbose when naming tests if it improves readability. `_ListOfTodoListsBelongingToUser` would suffice